internal/campaigns/changeset.go

```diff
@@ -194,6 +194,12 @@ func (s ChangesetCheckState) Valid() bool {
 	}
 }
 
+// CampaignChangeset stores the details on the connection between the two.
+type CampaignChangeset struct {
```

Comment:
One thought I had while reading through this PR was: do we really need the `Changeset` in the name? It seems that the type definition itself is independent of `Changeset` and might make for a more _telling_ name when read later:

```suggestion
// CampaignAssoc stores the details of a association to a Campaign.
type CampaignAssoc struct {
```

And then if you read `changeset.Campaigns` and see that it's a `[]CampaignAssoc` that makes total sense.